odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""Single-photon emission computerized tomography (SPECT) projectors """
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+try:
+    from NiftyPy import NiftyRec
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+from odl.operator import Operator
+from odl import uniform_discr_frompartition
+from odl.discr.lp_discr import DiscreteLp
+from odl.tomo.geometry.spect import ParallelHoleCollimatorGeometry
+
+
+__all__ = ('AttenuatedRayTransform', 'AttenuatedRayBackprojection',
+           'NIFTYREC_AVAILABLE')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class AttenuatedRayTransform(Operator):
+
+    """The discrete attenuated Ray transform between L^p spaces.
+
+    The attenuated Ray transform is a variant of the classical
+    Ray transform  in which functions are integrated over lines
+    with respect to an exponential weight given an attenuation map.
+    """
+
+    def __init__(self, dom, geometry, attenuation=None,
+                 impl='niftyrec', **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Domain of the forward projector, 3 dimensional.
+        geometry : `ParallelHoleCollimatorGeometry`
+            The geometry of the transform.
+        attenuation : domain `element-like`, optional
+            Linear attenuation map for SPECT.
+        impl : {'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec`_ tomography toolbox using the `NiftyPy`_
```

Comment:
Notation is 'astra_cpu'/'astra_cuda' for astra, perhaps we should decide on a style and use the same here.